numba/six.py

```diff
@@ -626,8 +627,15 @@ def byte2int(bs):
     def indexbytes(buf, i):
         return ord(buf[i])
     iterbytes = functools.partial(itertools.imap, ord)
-    import StringIO
-    StringIO = BytesIO = StringIO.StringIO
+    import StringIO as _StringIO
+    @contextlib.contextmanager
+    def StringIO(*args, **kwargs):
```

Comment:
This is changing what `StringIO` is (from type to function) and making it a contextmanager only.  Thus, one cannot do:

```python
s = StringIO()
print>>s, "showme"
```

Better to subclass from `_StringIO` and add `__enter__` and `__exit__`